components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -74,8 +78,101 @@ bool IsAllowContentSetting(HostContentSettingsMap* content_settings,
   return setting == CONTENT_SETTING_ALLOW;
 }
 
+// Sets only individual content settings for a given |resource_id|.
+void SetDefaultContentSettingForShield(Profile* profile,
+                                       const std::string& resource_id,
+                                       ShieldState state) {
+  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
+  map->SetContentSettingCustomScope(
+      ContentSettingsPattern::Wildcard(),
+      ContentSettingsPattern::Wildcard(),
+      CONTENT_SETTINGS_TYPE_PLUGINS,
+      resource_id,
+      state == ShieldState::kAllow ? CONTENT_SETTING_ALLOW
+                                   : CONTENT_SETTING_BLOCK);
+
+  if (resource_id == brave_shields::kCookies ||
+      resource_id == brave_shields::kFingerprinting) {
+    map->SetContentSettingCustomScope(
+        ContentSettingsPattern::Wildcard(),
+        ContentSettingsPattern::FromString("https://firstParty/*"),
+        CONTENT_SETTINGS_TYPE_PLUGINS,
+        resource_id,
+        state == ShieldState::kBlock ? CONTENT_SETTING_BLOCK
+                                     : CONTENT_SETTING_ALLOW);
+  } else {
+    DCHECK_NE(ShieldState::kBlock3rd, state);
+  }
+}
+
 }  // namespace
 
+void SetDefaultValueForShield(Profile* profile,
+                              const std::string& shield_id,
+                              ShieldState state) {
+  if (shield_id == kJavaScript) {
+    // Special case for scripts.
+    const bool allow = state == ShieldState::kAllow;
+    HostContentSettingsMapFactory::GetForProfile(profile)->
```

Comment:
can't we just call `SetDefaultContentSettingForShield` here?

Reply:
nope, because it is about `CONTENT_SETTINGS_TYPE_PLUGINS` and here we need `CONTENT_SETTINGS_TYPE_JAVASCRIPT`